rust/lance-encoding/src/decoder.rs

```diff
@@ -788,6 +788,12 @@ impl CoreFieldDecoderStrategy {
             let scheduler = self.create_primitive_scheduler(field, column_info, buffers)?;
             return Ok(scheduler);
         } else if data_type.is_binary_like() {
+            if column_infos.next().is_none() {
+                return Err(Error::InvalidInput {
+                    source: "column_info cannot be none".into(),
+                    location: location!(),
+                });
+            }
             let column_info = column_infos.next().unwrap().clone();
```

Comment:
Here's how you could avoid calling `.next()` twice:
```suggestion
            let Some(column_info) = column_infos.next() else {
                return Err(Error::InvalidInput {
                    source: "column_info cannot be none".into(),
                    location: location!(),
                });
            };
            let column_info = column_info.clone()
```

Reply:
Looking at the code above, I actually think we might instead be able to re-use this utility:
```suggestion
            let column_info = column_infos.expect_next()?;
```